### PR TITLE
Refactor startup scripts to be modular and reusable

### PR DESCRIPTION
**Pull request by shashidharatd**
Remove duplication of kubeadm startup scripts in every phase1 provider.
Moved the kubeadm startup scripts to phase2/kubeadm.

I have tested the changes in both gce and openstack and the changes do work for both of them.

/cc @jamiehannaford @pipejakob @cmluciano 
/assign @pipejakob

### CODE REVIEW

**Comment by shashidharatd** [comment]
/cc @dims @cmluciano 
request to review the changes commit by commit, so that its easy to perceive the changes.

File: phase1/openstack/configure-vm.sh [hunk 1]
```diff
@@ -23,3 +34,23 @@ apt-get update
 apt-get install -y docker-engine=1.12.0-0~xenial
 systemctl enable docker || true
 systemctl start docker || true
+
+ensure_gsutil_command(){
```
Comment (jamiehannaford) [inline phase1/openstack/configure-vm.sh]: why is this necessary for openstack?

File: phase2/kubeadm/configure-vm-kubeadm-master.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+
+# This is not meant to run on its own, but extends phase2/kubeadm/configure-vm-kubeadm.sh
+
+KUBEADM_DIR=/etc/kubeadm
+KUBEADM_CONFIG_FILE=$KUBEADM_DIR/kubeadm.yaml
+
+#TODO: we should probably be able to configure POD_NETWORK_CIDR from `make config` in future
+# and use the configured value by passing it on to CNI's. We resort to the below hard-coding
```
Comment (jamiehannaford) [inline phase2/kubeadm/configure-vm-kubeadm-master.sh]: nit: s/CNI's/CNIs/

File: phase2/kubeadm/configure-vm-kubeadm-master.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# since the current CNI's are not enabled to be configured with the user provided pod-network-cidr.
```
Comment (jamiehannaford) [inline phase2/kubeadm/configure-vm-kubeadm-master.sh]: yeah... can we create an issue that explains this?

File: phase2/kubeadm/configure-vm-kubeadm.sh [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    exit 1
+  elif [[ "$KUBEADM_VERSION" == "gs://"* ]]; then
+    KUBEADM_GS_DIR=$(echo $KUBEADM_VERSION | sed 's|/$||')
+    TMPDIR=/tmp/k8s-debs
+    mkdir $TMPDIR
+    gsutil cp "$KUBEADM_GS_DIR/kubeadm" $TMPDIR/kubeadm
+    cp $TMPDIR/kubeadm /usr/bin/kubeadm
+    rm -rf $TMPDIR
+  else
+    echo "Don't know how to handle kubeadm version: $KUBEADM_VERSION"
```
Comment (jamiehannaford) [inline phase2/kubeadm/configure-vm-kubeadm.sh]: how about something like `kubeadm version ($KUBEADM_VERSION) is not compatible with kubelet version ($KUBEADM_KUBELET_VERSION)`?

**Review by jamiehannaford — CHANGES_REQUESTED** [review]
looks good overall, just a few minor comments

File: phase1/openstack/configure-vm.sh (hunk 1 above)
Comment (shashidharatd) [reply phase1/openstack/configure-vm.sh]: gsutil is used to download the packages from GCS. GCS is the current location where all the release packages are hosted.

File: phase2/kubeadm/configure-vm-kubeadm-master.sh (hunk 2 above)
Comment (shashidharatd) [reply phase2/kubeadm/configure-vm-kubeadm-master.sh]: would fix.

File: phase2/kubeadm/configure-vm-kubeadm.sh (hunk 4 above)
Comment (shashidharatd) [reply phase2/kubeadm/configure-vm-kubeadm.sh]: I am not quite familiar with the whole logic, i just moved the files. probably @luxas/ @pipejakob is aware of this and can explain.

File: phase2/kubeadm/configure-vm-kubeadm-master.sh (hunk 3 above)
Comment (shashidharatd) [reply phase2/kubeadm/configure-vm-kubeadm-master.sh]: I will create an issue for this.

**Comment by cmluciano** [comment]
testing locally now

File: phase1/openstack/configure-vm.sh (hunk 1 above)
Comment (jamiehannaford) [reply phase1/openstack/configure-vm.sh]: ah okay, makes sense

File: phase1/openstack/configure-vm.sh [hunk 5]
```diff
@@ -5,6 +5,17 @@ set -o pipefail
 set -o nounset
 set -o xtrace
 
+KUBEADM_TOKEN="${k8s_kubeadm_token}"
+KUBEADM_VERSION="${k8s_kubeadm_version}"
+KUBEADM_KUBERNETES_VERSION="${k8s_kubeadm_kubernetes_version}"
+KUBEADM_KUBELET_VERSION="${k8s_kubeadm_kubelet_version}"
+KUBEADM_ENABLE_CLOUD_PROVIDER="${k8s_kubeadm_enable_cloud_provider}"
```
Comment (cmluciano) [inline phase1/openstack/configure-vm.sh]: This cloud-provider is not currently working. The config file also needs to be added

kubelet.service error
```
error: failed to run Kubelet: could not init cloud provider "openstack": no OpenStack cloud provider config
```

Comment (shashidharatd) [reply phase1/openstack/configure-vm.sh]: cloud-provider for openstack was not available even before this pr. So lets fix it in follow-up PR. @ss22ever interested in picking this up?

Comment (ss22ever) [reply phase1/openstack/configure-vm.sh]: @shashidharatd thanks, yes I would like to.

**Review by jamiehannaford — APPROVED** [review]

**Comment by cmluciano** [comment]
@pipejakob can we get this merged?

**Comment by bcreane** [comment]
Testing and merging of https://github.com/kubernetes/kubernetes/pull/55186 is blocked pending this PR getting merged. Would be great to get this in soon.

**Comment by cmluciano** [comment]
@shashidharatd Can you PR the test-infra repo and bump the sha similar to https://github.com/kubernetes/test-infra/pull/5275
